File: backend/app/schemas/analytics.py
# app/schemas/analytics.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from .products import CurrencyCode


class DealCostBreakdown(BaseModel):
    productCost: float = 0.0
    logisticsCost: float = 0.0
    dutiesTaxes: float = 0.0
    fxCost: float = 0.0
    commissions: float = 0.0
    otherCost: float = 0.0


class DealUnitEconomicsResult(BaseModel):
    dealId: str
    currency: CurrencyCode
    revenue: float
    totalCost: float
    grossMarginAbs: float
    grossMarginPct: float
    costBreakdown: DealCostBreakdown
    notes: Optional[str] = None

File: web/src/api/analytics.ts
// web/src/api/analytics.ts
import { api } from './client';
import type { AuthState } from '../state/authTypes';

export interface DealCostBreakdownDto {
  productCost: number;
  logisticsCost: number;
  dutiesTaxes: number;
  fxCost: number;
  commissions: number;
  otherCost: number;
}

export interface DealUnitEconomicsDto {
  dealId: string;
  currency: 'RUB' | 'CNY' | 'USD';
  revenue: number;
  totalCost: number;
  grossMarginAbs: number;
  grossMarginPct: number;
  costBreakdown: DealCostBreakdownDto;
  notes?: string | null;
}

export async function getDealUnitEconomics(
  auth: AuthState,
  dealId: string,
): Promise<DealUnitEconomicsDto> {
  return api<DealUnitEconomicsDto>(
    `/analytics/deals/${dealId}/unit-economics`,
    {},
    auth.tokens.accessToken,
  );
}